tony-core/src/main/java/com/linkedin/tony/events/EventHandler.java

```diff
@@ -40,18 +40,23 @@ public EventHandler(FileSystem fs, BlockingQueue<Event> q) {
     myFs = fs;
   }
 
-  public void setUpThread(Path intermDir, JobMetadata metadata) {
+  public boolean setUpThread(Path intermDir, JobMetadata metadata) {
     if (intermDir == null) {
-      return;
+      return true;
     }
     inProgressHistFile = new Path(intermDir, HistoryFileUtils.generateFileName(metadata));
     try {
       out = myFs.create(inProgressHistFile);
       dataFileWriter.create(Event.SCHEMA$, out);
       LOG.info("Writing events to " + inProgressHistFile);
+    } catch (NoSuchMethodError e) {
```

Comment:
what happens if an exception is raised that's neither of these 2(NoSuchMethodError or IOException)? As much as it is an antipattern, should we return false for all `Exception` throws?

Reply:
I'm little conservative with catching exceptions (legacy). I would prefer to catch exceptions that we can properly deal with otherwise just let it bubble up.